### ISSUE TITLE
Loss decreases for the first 0.2 epoch before increasing

### ISSUE THREAD

**Issue by shuyijia** [facebookresearch/crystal-llm#6]
Thanks for the good work!

Tried training a 13B model and the loss decreased from ~1.3 to around 0.5 before increasing to ~6, at which it hovered and the grad norm became zero. Am I missing something here?

Selected log lines:

{'loss': 1.3587, 'grad_norm': 0.24448351562023163, 'learning_rate': 5e-05, 'epoch': 0.0}
...
{'loss': 0.4827, 'grad_norm': 0.27102014422416687, 'learning_rate': 0.0004999954015022048, 'epoch': 0.11}
...
{'loss': 0.5813, 'grad_norm': 1.3563207387924194, 'learning_rate': 0.0004999867104183737, 'epoch': 0.17}
{'loss': 0.7004, 'grad_norm': 1.2771899700164795, 'learning_rate': 0.0004999863166778732, 'epoch': 0.17}
{'loss': 0.6607, 'grad_norm': 17.22315788269043, 'learning_rate': 0.0004999859171895474, 'epoch': 0.17}
{'loss': 1.0354, 'grad_norm': 7.1172919273376465, 'learning_rate': 0.0004999855119534057, 'epoch': 0.17}
{'loss': 0.6606, 'grad_norm': 21.806798934936523, 'learning_rate': 0.0004999851009694573, 'epoch': 0.17}
{'loss': 0.6845, 'grad_norm': 23.015594482421875, 'learning_rate': 0.0004999846842377115, 'epoch': 0.18}
{'loss': 0.8643, 'grad_norm': 3.768519639968872, 'learning_rate': 0.000499984261758178, 'epoch': 0.18}
{'loss': 0.8247, 'grad_norm': 14.410096168518066, 'learning_rate': 0.0004999838335308665, 'epoch': 0.18}
{'loss': 0.8874, 'grad_norm': 7.4343671798706055, 'learning_rate': 0.000499983399555787, 'epoch': 0.18}
{'loss': 1.1015, 'grad_norm': 3.0113959312438965, 'learning_rate': 0.0004999829598329492, 'epoch': 0.18}
{'loss': 1.156, 'grad_norm': 20.571786880493164, 'learning_rate': 0.0004999825143623633, 'epoch': 0.19}
{'loss': 1.4336, 'grad_norm': 3.120924711227417, 'learning_rate': 0.0004999820631440396, 'epoch': 0.19}
{'loss': 1.9073, 'grad_norm': 1.6686218976974487, 'learning_rate': 0.0004999816061779886, 'epoch': 0.19}
{'loss': 1.7766, 'grad_norm': 14.47624397277832, 'learning_rate': 0.0004999811434642206, 'epoch': 0.19}
{'loss': 2.5124, 'grad_norm': 12.572964668273926, 'learning_rate': 0.0004999806750027462, 'epoch': 0.2}
{'loss': 1.9898, 'grad_norm': 6.252790451049805, 'learning_rate': 0.0004999802007935763, 'epoch': 0.2}
{'loss': 1.9474, 'grad_norm': 0.626639723777771, 'learning_rate': 0.0004999797208367218, 'epoch': 0.2}
{'loss': 1.8815, 'grad_norm': 11.334622383117676, 'learning_rate': 0.0004999792351321938, 'epoch': 0.2}
{'loss': 1.9177, 'grad_norm': 0.673676609992981, 'learning_rate': 0.0004999787436800033, 'epoch': 0.2}
{'loss': 1.9395, 'grad_norm': 4.5425333976745605, 'learning_rate': 0.0004999782464801618, 'epoch': 0.21}
{'loss': 2.0561, 'grad_norm': 3.543849229812622, 'learning_rate': 0.0004999777435326805, 'epoch': 0.21}
...
{'loss': 5.9967, 'grad_norm': 0.01927277259528637, 'learning_rate': 0.0004999089434963053, 'epoch': 0.4}
...
{'loss': 6.1372, 'grad_norm': 0.0, 'learning_rate': 0.0004977269022315838, 'epoch': 1.91}

**Comment by ngruver** [comment]
This is odd, I wouldn't expect the loss to increase significantly and certainly not above 1. What batch size and how many GPUs are you using?

**Comment by shuyijia** [comment]
I trained a 13B model with batch size of 4 on five GPUs. I used the same learning rate (0.0005) as reported in the paper. 

Previously, I have tried training a 7B model with batch size of 1 on a single GPU with the default learning rate (1e-4) in the argparse and observed the same pattern: loss decreased from 1-ish to 0.5ish, before increasing to 6-7 and did not decrease after that.

**Comment by ngruver** [comment]
Sorry you've had trouble with the training code. I will look into the problem this weekend by directly rerunning some things. It's possible some problems were introduced when porting the code over to this repo.
